packages/aggregator/src/tests/analytics/aggregateAnalytics/aggregations/sumAcrossPeriods.test.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import { sumAcrossPeriods } from '../../../../analytics/aggregateAnalytics/aggregations/sumAcrossPeriods';
+
+describe('sumAcrossPeriods', () => {
+  const arrayToAnalytics = arrayAnalytics =>
+    arrayAnalytics.map(([dataElement, organisationUnit, period, value]) => ({
+      dataElement,
+      organisationUnit,
+      period,
+      value,
+    }));
+
+  const ANALYTICS = arrayToAnalytics([
+    // BCD01 - TO
+    ['BCD01', 'TO', '201601', 1],
+    ['BCD01', 'TO', '201801', 0.1],
+    ['BCD01', 'TO', '201901', 0.01],
+    // BCD01 - PG
+    ['BCD01', 'PG', '201601', 2],
+    ['BCD01', 'PG', '201801', 0.2],
+    ['BCD01', 'PG', '201901', 0.02],
+    // BCD02 - TO
+    ['BCD02', 'TO', '201801', 4],
+    ['BCD02', 'TO', '202001', 0.44],
+  ]);
+
+  const CURRENT_DATE_STUB = new Date(2020, 1, 1);
+  const CURRENT_PERIOD_STUBS = {
+    DAY: '20200131',
+    YEAR: '2020',
+  };
+
+  let clock;
+
+  before(() => {
+    clock = sinon.useFakeTimers({ now: CURRENT_DATE_STUB });
+  });
+
+  after(() => {
+    clock.restore();
+  });
+
+  it('single analytic', () => {
+    expect(sumAcrossPeriods([ANALYTICS[0]])).to.have.same.deep.members([ANALYTICS[0]]);
+  });
+
+  it('multiple analytics', () => {
+    expect(sumAcrossPeriods(ANALYTICS)).to.have.same.deep.members(
+      arrayToAnalytics([
+        ['BCD01', 'TO', '201601', 1.11],
+        ['BCD01', 'PG', '201601', 2.22],
+        ['BCD02', 'TO', '201801', 4.44],
```

Comment:
The tests revealed a flaw in the current functionality!! We should use the max period among the summed analytics. It should be:

```suggestion
        ['BCD01', 'TO', '201901', 1.11],
        ['BCD01', 'PG', '201901', 2.22],
        ['BCD02', 'TO', '202001', 4.44],
```

I decided to keep this PR regression-free and not fix this now

Reply:
Good catch unit tests!